js/bootstrap-colorpicker-module.js

```diff
@@ -1,574 +1,663 @@
-              colorpickerPreview = colorpickerTemplate.find('colorpicker-preview'),
-              pickerColorPointers = colorpickerTemplate.find('i');
-
-          $compile(colorpickerTemplate)($scope);
-
-          if (withInput) {
-            var pickerColorInput = colorpickerTemplate.find('input');
-            pickerColorInput
-                .on('mousedown', function(event) {
-                  event.stopPropagation();
-                })
-                .on('keyup', function(event) {
-                  var newColor = this.value;
-                  elem.val(newColor);
-                  if(ngModel) {
-                    $scope.$apply(ngModel.$setViewValue(newColor));
-                  }
-                  event.stopPropagation();
-                  event.preventDefault();
+    .factory('Swatches', function() {
+        'use strict';
+
+        // Contains all the preset colors
+        var colors = [
+            ['#1e5667', '#3d7e92', '#97cad9', '#b6dfeb'],
+            ['#863a27', '#c75336', '#d8765d', '#fbbfb0'],
+            ['#677b31', '#8ba446', '#b0c870', '#def0ac'],
+            ['#f28e1b', '#e19d4e', '#ffc17a', '#f8e7b9'],
+            ['#520039', '#854271', '#a186be', '#d9cce7'],
+            ['#603816', '#ad6628', '#e0b48d', '#edd1b8']
+        ];
+        return {
+            getSwatches: function(elem) {
+                return colors;
+            }
+        };
+    })
+    .directive('colorpicker', ['$window', '$document', '$compile', '$timeout', 'Color', 'Slider', 'Helper', 'Swatches', function($window, $document, $compile, $timeout, Color, Slider, Helper, Swatches) {
+        'use strict';
+        return {
+            require: '?ngModel',
+            restrict: 'A',
+            link: function($scope, elem, attrs, ngModel) {
+                var
+                    streamid = attrs.streamid,
+                    thisFormat = attrs.colorpicker ? attrs.colorpicker : 'hex',
+                    position = angular.isDefined(attrs.colorpickerPosition) ? attrs.colorpickerPosition : 'right',
+                    inline = angular.isDefined(attrs.colorpickerInline) ? attrs.colorpickerInline : false,
+                    fixedPosition = angular.isDefined(attrs.colorpickerFixedPosition) ? attrs.colorpickerFixedPosition : false,
+                    target = angular.isDefined(attrs.colorpickerParent) ? elem.parent() : angular.element(document.body),
+                    withInput = angular.isDefined(attrs.colorpickerWithInput) ? attrs.colorpickerWithInput : false,
+                    inputTemplate = withInput ? '<input type="text" name="colorpicker-input">' : '',
+                    switchButton = !inline ? '<div ng-if="!palette" class="switch-colorpicker" ng-click="switchColorpickerView($event);" stop-propagation><i class="icon icon-paintbrush"></i></div><div class="switch-colorpicker" ng-if="palette" ng-click="switchColorpickerView($event);" stop-propagation><i class="icon icon-swatches"></i></div>' : '',
```

Comment:
Will the `inline` flag ever be triggered here? Since this has been modified heavily for our usecase, should we just remove the check?